railties/lib/rails/tasks/framework.rake

```diff
@@ -66,3 +68,24 @@ namespace :rails do
     end
   end
 end
+
+namespace :rails do
+  Rake.application.top_level_tasks.grep(/^rails:/).each do |task|
+    ActiveSupport::Deprecation.warn(<<-MSG.squish)
+      Running this task with the rails task namespace is deprecated in favor of app.
+      Please run e.g. bin/rails #{task.sub(/^rails:/, 'app:')} instead."
+    MSG
+  end
```

Comment:
Sorry wasn't clear enough.

> Don't need to print deprecation warnings for every task in the namespace.

I meant we shouldn't use `Rake.application.top_level_tasks.grep(/^rails:/)`. I didn't mean to remove the deprecation enrichment of the `rails:` namespaced tasks, that part was correct before.